wpimath/src/main/native/include/frc/proto/MatrixProto.inc

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include <wpi/ProtoHelper.h>
+
+#include "frc/proto/MatrixProto.h"
+#include "wpimath.pb.h"
+
+template <int Rows, int Cols, int Options, int MaxRows, int MaxCols>
+  requires(Cols != 1)
+google::protobuf::Message*
+wpi::Protobuf<frc::Matrixd<Rows, Cols, Options, MaxRows, MaxCols>>::New(
+    google::protobuf::Arena* arena) {
+  return wpi::CreateMessage<wpi::proto::ProtobufMatrix>(arena);
+}
+
+template <int Rows, int Cols, int Options, int MaxRows, int MaxCols>
+  requires(Cols != 1)
+frc::Matrixd<Rows, Cols, Options, MaxRows, MaxCols>
+wpi::Protobuf<frc::Matrixd<Rows, Cols, Options, MaxRows, MaxCols>>::Unpack(
+    const google::protobuf::Message& msg) {
+  auto m = static_cast<const wpi::proto::ProtobufMatrix*>(&msg);
+  if (m->num_rows() != Rows || m->num_cols() != Cols) {
+    // TODO Error
+  }
+  if (m->data_size() != Rows * Cols) {
+    // TODO Error
```

Comment:
I think these should just ignore the incoming data if it's incorrectly sized.

Reply:
I don't know what ignoring the incoming data looks like, since we have to return a result, so I guess we throw. Should we define a custom exception type to throw or use a standard exception type?